app/src/lib/stores/app-store.ts

```diff
@@ -4736,19 +4752,39 @@ export class AppStore extends TypedBaseStore<IAppState> {
     await createDesktopStashEntry(repository, branchName)
   }
 
-  public async _popStash(repository: Repository, branch: Branch) {
-    const stash = await getLastDesktopStashEntryForBranch(
-      repository,
-      branch.name
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public async _popStashEntry(repository: Repository, stashEntry: IStashEntry) {
+    const gitStore = this.gitStoreCache.get(repository)
+    await gitStore.performFailableOperation(() => {
+      return popStashEntry(repository, stashEntry.stashSha)
+    })
+    log.info(
+      `[AppStore. _popStashEntry] popped stash with commit id ${
+        stashEntry.stashSha
+      }`
     )
 
-    if (stash === null) {
-      log.warn('no stash found that matches')
-      return
-    }
+    await this._refreshRepository(repository)
+
+    this.emitUpdate()
```

Comment:
I think this one can also be dropped:

```suggestion
```

Reply:
removed in https://github.com/desktop/desktop/pull/7213/commits/3c9e4335dec0ae98f170b867f562450da439f783